libs/datasets/manual_filter.py

```diff
@@ -153,23 +153,42 @@ def touched_subset(
     ds_in: timeseries.MultiRegionDataset, ds_out: timeseries.MultiRegionDataset
 ) -> timeseries.MultiRegionDataset:
     """Given the input and output of `run`, returns the time series from ds_in that have tags in
-    ds_out added by `run` combined with tags of these time series copied from ds_out."""
-    # Expected tags do not appear in the input
+    ds_out added by `run` and dropped observations, combined with tags of these time series
+    copied from ds_out."""
+    # Check that tags added by `run` do not appear in the input of `run`.
     assert ds_in.tag.index.unique(taglib.TagField.TYPE).intersection(_EXPECTED_TYPES).empty
     tag_mask = ds_out.tag.index.get_level_values(taglib.TagField.TYPE).isin(_EXPECTED_TYPES)
-    # touch_index identifies the subset of time series rows that have an _EXPECTED_TYPES tag.
-    touch_index = ds_out.tag.index[tag_mask].droplevel(taglib.TagField.TYPE).unique()
+    # ts_has_tag_index identifies the subset of time series rows that have an _EXPECTED_TYPES tag.
+    ts_has_tag_index = ds_out.tag.index[tag_mask].droplevel(taglib.TagField.TYPE).unique()
     wide_dates_in_df = ds_in.timeseries_bucketed_wide_dates
-    assert wide_dates_in_df.index.names == touch_index.names
-    assert touch_index.isin(wide_dates_in_df.index).all()
-    # Get all ds_in.tag associated with any time series in touch_index
-    tag_in = ds_in.tag.loc[ds_in.tag.index.droplevel(taglib.TagField.TYPE).isin(touch_index)]
-    # Get only the ds_out.tag where the type is _EXPECTED_TYPES. These are the tags added by
-    # `run` and not in ds_in.tag.
-    tag_out = ds_out.tag.loc[tag_mask]
+    assert wide_dates_in_df.index.names == ts_has_tag_index.names
+    assert ts_has_tag_index.isin(wide_dates_in_df.index).all()
+    # For time series identified by ts_has_tag_index find the number of observations in ds_in and
+    # ds_out.
+    ts_in_observations = wide_dates_in_df.reindex(ts_has_tag_index).notna().sum(axis="columns")
+    ts_out_observations = (
+        ds_out.timeseries_bucketed_wide_dates.reindex(ts_has_tag_index).notna().sum(axis="columns")
+    )
+    # Some time series may have been dropped from ds_in to ds_out (if all observations were
+    # dropped) but no time series were added.
+    assert ts_out_observations.index.isin(ts_in_observations.index).all()
```

Comment:
Since we explicitly `reindex()`ed both of these based on `ts_has_tag_index` it seems like these indexes should always be exactly the same (and therefore the assert probably is unnecessary), no?

Reply:
Good point. Removed and updated comment.